src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToCopyLocalDependencies.cs

```diff
@@ -0,0 +1,153 @@
+                TargetFrameworks = "netcoreapp3.0",
+                IsExe = true
+            };
+
+            testProject.PackageReferences.Add(new TestPackageReference("Newtonsoft.Json", "11.0.2"));
+            testProject.PackageReferences.Add(new TestPackageReference("sqlite", "3.13.0"));
+
+             var testProjectInstance = _testAssetsManager
+                .CreateTestProject(testProject)
+                .Restore(Log, ProjectName);
+
+            var buildCommand = new BuildCommand(Log, testProjectInstance.TestRoot, ProjectName);
+
+            buildCommand.Execute("-p:CopyLocalLockFileAssemblies=false")
+                .Should()
+                .Pass();
+
+            var outputDirectory = buildCommand.GetOutputDirectory(testProject.TargetFrameworks);
+            outputDirectory.Should().OnlyHaveFiles(new[] {
+                $"{ProjectName}{Constants.ExeSuffix}",
+                $"{ProjectName}.deps.json",
+                $"{ProjectName}.dll",
+                $"{ProjectName}.pdb",
+                $"{ProjectName}.runtimeconfig.dev.json",
+                $"{ProjectName}.runtimeconfig.json",
+            });
+        }
+
+        [Fact]
+        public void It_copies_local_specific_runtime_package_dependencies_on_build()
+        {
+            const string ProjectName = "TestProjWithPackageDependencies";
+
+            var rid = EnvironmentInfo.GetCompatibleRid("netcoreapp3.0");
+
+            TestProject testProject = new TestProject()
+            {
+                Name = ProjectName,
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp3.0",
+                IsExe = true
+            };
+
+            testProject.AdditionalProperties.Add("RuntimeIdentifier", rid);
+            testProject.AdditionalProperties.Add("SelfContained", "false");
+            testProject.PackageReferences.Add(new TestPackageReference("Newtonsoft.Json", "11.0.2"));
+            testProject.PackageReferences.Add(new TestPackageReference("sqlite", "3.13.0"));
+
+             var testProjectInstance = _testAssetsManager
+                .CreateTestProject(testProject)
+                .Restore(Log, ProjectName);
+
+            var buildCommand = new BuildCommand(Log, testProjectInstance.TestRoot, ProjectName);
+
+            buildCommand.Execute()
+                .Should()
+                .Pass();
+
+            var outputDirectory = buildCommand.GetOutputDirectory(targetFramework: testProject.TargetFrameworks, runtimeIdentifier: rid);
+
+            outputDirectory.Should().OnlyHaveFiles(new[] {
+                $"{ProjectName}{Constants.ExeSuffix}",
+                $"{ProjectName}.deps.json",
+                $"{ProjectName}.dll",
+                $"{ProjectName}.pdb",
+                $"{ProjectName}.runtimeconfig.dev.json",
+                $"{ProjectName}.runtimeconfig.json",
+                "Newtonsoft.Json.dll",
+                $"{FileConstants.DynamicLibPrefix}sqlite3{FileConstants.DynamicLibSuffix}",
```

Comment:
What's going to happen if we run this test on a RID that the SQLite package doesn't have an asset for, e.g. MUSL Linux?  I think this test would fail.  That is probably fine, we can deal with that when we come across it, but a comment about that might be helpful.

Reply:
I'll add a comment regarding this potential future breakage.